File: asgn2/python_code/task4_java_submit/DiningHall.java
package assg2task4;
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/*
 * CSCI3180 Principles of Programming Languages
 *
 * --- Declaration ---
 *
 * I declare that the assignment here submitted is original except for source
 * material explicitly acknowledged. I also acknowledge that I am aware of
 * University policy and regulations on honesty in academic work, and of the
 * disciplinary guidelines and procedures applicable to breaches of such policy
 * and regulations, as contained in the website
 * http://www.cuhk.edu.hk/policy/academichonesty/
 *
 * Assignment 2
 * Name : Mao Yuxuan
 * Student ID : 1155062018
 * Email Addr : yxmao5@cse.cuhk.edu.hk
 */

/**
 *
 * @author kenchan
 */
public class DiningHall extends Room {
    protected double dinner_price;
    public DiningHall(String Name, int Position, double Dinner_price){
        super(Name, Position);
        this.dinner_price = Dinner_price;
    }
    public void ServerDinnerTo(Attendant Obj){
        if (Obj.GetPosition()==this.position){
            if (Obj.Pay(this.dinner_price)>=0){
                System.out.println(Obj.GetName() + " finished the dinner.");
            }
            else{
                System.out.println(Obj.GetName() + " didn't have enough money.");
            }
        }
        else{
            System.out.println(Obj.GetName() + " was too far away to be served.");
        }
    }
}

File: Programming_Languages/asgn2/python_code/task3_python/DiningHall.py
# this is the python Dinning hall code
import Attendant
import DumbFixture
import Room
import SmartFixture

class DiningHall(Room):
    def __init__(self,Name,Position,Dinner_price):
        super(DiningHall,self).__init__(Name,Position)
        self._dinner_price=Dinner_price

    def ServerDinnerTo(self,Obj):
        if(Obj.GetPosition()==self._position):
            if(Obj.Pay(self._dinner_price)>=0):
                print ('%s finished the dinner.' % Obj.GetName())
            else:
                print ("%s didn't have enough money." % Obj.GetName())
        else:
            print ('%s was too far away to be served.' % Obj.GetName())

